cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -126,6 +122,135 @@ impl LockedManifest {
         Ok(ContainerBuilder::new(container_builder_path))
     }
 
+    pub fn read_from_file(path: &CanonicalPath) -> Result<Self, LockedManifestError> {
+        let contents = fs::read(path).map_err(LockedManifestError::ReadLockfile)?;
+        serde_json::from_slice(&contents).map_err(LockedManifestError::ParseLockfile)
+    }
+
+    pub fn check_lockfile(
+        path: &CanonicalPath,
+    ) -> Result<Vec<LockfileCheckWarning>, LockedManifestError> {
+        let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
+        pkgdb_cmd
+            .args(["manifest", "check"])
+            .arg("--lockfile")
+            .arg(path.as_os_str());
+
+        debug!("checking lockfile with command: {}", pkgdb_cmd.display());
+
+        let value = call_pkgdb(pkgdb_cmd).map_err(LockedManifestError::CheckLockfile)?;
+        let warnings: Vec<LockfileCheckWarning> =
+            serde_json::from_value(value).map_err(LockedManifestError::ParseCheckWarnings)?;
+
+        Ok(warnings)
+    }
+}
+
+impl ToString for LockedManifest {
+    fn to_string(&self) -> String {
+        serde_json::json!(self).to_string()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LockedManifestCatalog {
+    #[serde(rename = "lockfile-version")]
+    version: Version<1>,
+    /// original manifest that was locked
+    manifest: TypedManifestCatalog,
+    /// locked groups
+    groups: Vec<LockedGroup>,
```

Comment:
we discussed to store the resolved packages flat.
However, packages miss a `system` attribute in allowing to properly filter out the packages for the current system.
This is relevant both for listing and building

Reply:
Added a comment to this ticket to address this.  If there's additional comments or discussions, lets have them there and we can get this in. https://github.com/flox/catalog-server/issues/32

Reply:
Resolving this as closed, BUT if there is work here to be updated once that other change is complete, lets add a ticket for that as dependent of the one with the catalog-server change flox/catalog-server#32 .